ldap/servers/plugins/dynamic_lists/dynamic_lists.c

```diff
@@ -0,0 +1,301 @@
+                          "dynamic_lists_pre_entry - %s - Missing filter in LDAP URL %s\n",
+                          slapi_entry_get_dn_const(list_entry), url);
+            continue;
+        }
+
+        if (ludp->lud_dn == NULL) {
+            slapi_log_err(SLAPI_LOG_ERR, DYNAMIC_LISTS_PLUGIN_SUBSYSTEM,
+                          "dynamic_lists_pre_entry - %s - Missing base dn in LDAP URL %s\n",
+                          slapi_entry_get_dn_const(list_entry), url);
+            continue;
+        }
+
+        if (ludp->lud_attrs != NULL && ludp->lud_attrs[0] != NULL) {
+            /* If an attribute is specified use it */
+            list_attr = ludp->lud_attrs[0];
+        }
+
+        /*
+        * Check if the client has permission to read the list attribute
+        */
+        if (dynamic_lists_check_access(pb, list_entry,
+                                       list_attr ? list_attr : listConfig.attr) == false)
+        {
+            /* Access denied, so we can't proceed */
+            continue;
+        }
+
+        slapi_log_err(SLAPI_LOG_PLUGIN, DYNAMIC_LISTS_PLUGIN_SUBSYSTEM,
+                      "dynamic_lists_pre_entry - %s (url: %s) - Performing internal search\n",
+                      slapi_entry_get_dn_const(list_entry), url);
+
+        /*
+        * Get our DN's groupOfUrls attribute
+        */
+        search_pb = slapi_pblock_new();
+        slapi_search_internal_set_pb(search_pb, ludp->lud_dn, ludp->lud_scope, ludp->lud_filter,
+                                     NULL, 0, NULL, NULL, dynamic_lists_get_plugin_id(), 0);
+        slapi_search_internal_pb(search_pb);
+        slapi_pblock_get(search_pb, SLAPI_PLUGIN_INTOP_RESULT, &rc);
+        if (LDAP_SUCCESS != rc) {
+            /* log an error and use the plugin entry for the config */
+            slapi_log_err(SLAPI_LOG_ERR, DYNAMIC_LISTS_PLUGIN_SUBSYSTEM,
+                          "dynamic_lists_pre_entry - "
+                          "%s - Internal search based on LDAP url (%s) failed: err=%d\n",
+                          slapi_entry_get_dn_const(list_entry), url, rc);
+            continue;
+        } else {
+            list_set = slapi_valueset_new();
+            slapi_pblock_get(search_pb, SLAPI_PLUGIN_INTOP_SEARCH_ENTRIES, &entries);
+            for (size_t i = 0; entries && entries[i]; i++) {
+                if (list_attr == NULL) {
+                    /* No list attribute, so use the DN of the entry */
+                    const char *dn = slapi_entry_get_dn_const(entries[i]);
+                    slapi_valueset_add_value_ext(list_set, slapi_value_new_string(dn),
+                                                 SLAPI_VALUE_FLAG_PASSIN);
+                } else {
+                    /* Use an attribute/value from within the entry */
+                    const char *entry_list_attr = slapi_entry_attr_get_ref(entries[i],
+                                                                           list_attr ? list_attr : listConfig.attr);
```

Comment:
I think we don't need the ternary operator `list_attr ? list_attr : listConfig.attr` here as we're already in the `else` block where `list_attr != NULL`.